src/app/core/tasks/threadpool.nim

```diff
@@ -1,260 +1,45 @@
-import # std libs
-  atomics, json, sequtils, tables
+import #std libs
+  std/threadpool
 
 import # vendor libs
-  chronicles, chronos, json_serialization, task_runner
+  json_serialization, json, chronicles
 
 import # status-desktop libs
-  ./common
+  ./common 
 
-export
-  chronos, common, json_serialization
+export common, json_serialization
 
 logScope:
   topics = "task-threadpool"
-
+  
 type
   ThreadPool* = ref object
-    chanRecvFromPool: AsyncChannel[ThreadSafeString]
-    chanSendToPool: AsyncChannel[ThreadSafeString]
-    thread: Thread[PoolThreadArg]
-    size: int
-    running*: Atomic[bool]
-  PoolThreadArg = object
-    chanSendToMain: AsyncChannel[ThreadSafeString]
-    chanRecvFromMain: AsyncChannel[ThreadSafeString]
-    size: int
-  TaskThreadArg = object
-    id: int
-    chanRecvFromPool: AsyncChannel[ThreadSafeString]
-    chanSendToPool: AsyncChannel[ThreadSafeString]
-  ThreadNotification = object
-    id: int
-    notice: string
-
-# forward declarations
-proc poolThread(arg: PoolThreadArg) {.thread.}
 
-const MaxThreadPoolSize = 16
+const MinThreadPoolSize = 16
 
-proc init(self: ThreadPool) =
-  self.chanRecvFromPool.open()
-  self.chanSendToPool.open()
-  let arg = PoolThreadArg(
-    chanSendToMain: self.chanRecvFromPool,
-    chanRecvFromMain: self.chanSendToPool,
-    size: self.size
-  )
-  createThread(self.thread, poolThread, arg)
-  # block until we receive "ready"
-  discard $(self.chanRecvFromPool.recvSync())
+proc init*(self: ThreadPool) = 
+  setMinPoolSize(MinThreadPoolSize)
```

Comment:
Do we really need to set it, why not default? [Threadpool can adapt](https://github.com/nim-lang/Nim/blob/80e739f2bf5a3e40df7000566a9cbb9cfab29ac4/lib/pure/concurrency/cpuload.nim#L36) the number of (windows and linux) threads based on CPU load.

```suggestion
proc init*(self: ThreadPool) = 
  when defined(macosx):
    setMinPoolSize(16)
```

Reply:
I don't think we need to set it, but I did it this way to avoid false positive results regarding memory usage.
It looks like the app doesn't actually need all these threads. Haven't seen more than 2 threads being used in normal conditions.